receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -136,6 +110,38 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
 	if err != nil {
 		return err
 	}
+	if c.consumerGroup == nil {
```

Comment:
add a comment about this, sounds kind of random to check for this, but it is for testing.

Reply:
It's not just for testing, so I guess we can make this more obvious.
We have had panics on shutdown when the collector doesn't start correctly, where shutdown might get called on receivers that have not started yet. This check ensures that we don't crash if `Shutdown` is called without calling `Start` first.

Reply:
Are you sure? Because this is in `Start`

Reply:
oh sorry you're right. I'll add a comment.

Reply:
done